cmd/disk-cache.go

```diff
@@ -184,20 +190,25 @@ func (c *cacheObjects) GetObjectNInfo(ctx context.Context, bucket, object string
 
 	cacheReader, cacheErr := c.get(ctx, dcache, bucket, object, rs, h, opts)
 	if cacheErr == nil {
+		cacheObjSize = cacheReader.ObjInfo.Size
+		if rs != nil {
+			if _, len, err := rs.GetOffsetLength(cacheObjSize); err == nil {
+				cacheObjSize = len
+			}
+		}
 		cc = cacheControlOpts(cacheReader.ObjInfo)
-		if !cc.isStale(cacheReader.ObjInfo.ModTime) {
+		// Note that the below check is to avoid doing a HEAD if Cache-Control
+		// headers are set.
+		if !cc.isEmpty() && !cc.isStale(cacheReader.ObjInfo.ModTime) {
```

Comment:
@poornas when `cache-control` is not set, the `incCacheStats` will never be called even if this is a cache hit.

Reply:
@nitisht,when cc is not set, cache hits should be incremented further down in this codeblock after ETag verification is done. See [line 236](https://github.com/minio/minio/pull/8617/files#diff-9a7adf3c9b3149ed30ea4c51b712b87eR236)